### PR TITLE
Modify rule S6973: Adding PyTorch optimizers

### PR DESCRIPTION
**Pull request by joke1196**
## Review

A dedicated reviewer checked the rule description successfully for:

- [ ] logical errors and incorrect information
- [ ] information gaps and missing content
- [ ] text style and tone
- [ ] PR summary and labels follow [the guidelines](https://github.com/SonarSource/rspec/#to-modify-an-existing-rule)

### CODE REVIEW

File: rules/S6973/python/how-to-fix-it/pytorch.adoc [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+----
+
+==== Compliant solution
+
+[source,python,diff-id=2,diff-type=compliant]
+----
+from my_model import model
+from torch.optim import Adadelta
+
+optimizer = Adadelta(model.parameters(), lr = 0.001, weight_decay = 0.003) # Compliant
```
Comment (ghislainpiot) [inline rules/S6973/python/how-to-fix-it/pytorch.adoc:24]: nit: We might want to switch the exemple to Adam/AdamW to make it more relatable to users ?

File: rules/S6973/python/rule.adoc [hunk 2]
```diff
@@ -1,18 +1,16 @@
-This rule raises an issue when a Scikit-learn model is instantiated without specifying the important hyperparameters. 
-
-
+This rule raises an issue when a machine learning model or optimizer is instantiated without specifying the important hyperparameters.
```
Comment (ghislainpiot) [inline rules/S6973/python/rule.adoc:1]: I'm sorry I didn't catch this last time, but I don't think we should call it a "model," especially now that we are including Pytorch in the rule.

File: rules/S6973/python/rule.adoc [hunk 3]
```diff
@@ -1,18 +1,16 @@
...
 
-When instantiating a Scikit-learn estimator, it will use default values for the hyperparameters that are not specified.
-Relying on the default values can lead to non-reproducible results across diffferent versions of the library. 
+When instantiating an estimator or an optimizer, it will use default values for the hyperparameters that are not specified.
+Relying on the default values can lead to non-reproducible results across different versions of the library. 
 
 Furthermore, the default values might not be the best choice for the specific problem at hand and can lead to suboptimal performance.
 
 Here are the estimators and the parameters considered by this rule : 
 [cols="1,1"]
```
Comment (ghislainpiot) [inline rules/S6973/python/rule.adoc:11]: I am also sorry for not catching this last time, but I find the table hard to read. It's the same for the `numpy.Generator` rule. I don't know how we could fix it however.

**Review by ghislainpiot — CHANGES_REQUESTED** [review]
The bulk of the rule is good, it is just small details and things I did not catch in the previous sprint.

File: rules/S6973/python/rule.adoc (hunk 3 above)
Comment (joke1196) [reply rules/S6973/python/rule.adoc]: I am not sure we can do much as some formatting features are disabled it seems, i.e. striping even or odd rows

**Review by ghislainpiot — APPROVED** [review]
LGTM!

File: rules/S6973/python/rule.adoc (hunk 3 above)
Comment (jean-jimbo-sonarsource) [reply rules/S6973/python/rule.adoc]: I'll follow up on formatting tables.

File: rules/S6973/python/rule.adoc [hunk 4]
```diff
@@ -1,90 +1,66 @@
-This rule raises an issue when a Scikit-learn model is instantiated without specifying the important hyperparameters. 
-
-
+This rule raises an issue when a machine learning estimator or optimizer is instantiated without specifying the important hyperparameters. 
 
 == Why is this an issue?
 
-When instantiating a Scikit-learn estimator, it will use default values for the hyperparameters that are not specified.
-Relying on the default values can lead to non-reproducible results across diffferent versions of the library. 
+When instantiating an estimator or an optimizer, it will use default values for the hyperparameters that are not specified.
```
Comment (jean-jimbo-sonarsource) [inline rules/S6973/python/rule.adoc:5]: Suggestion: "When instantiating an estimator or an optimizer, default values for any hyperparameters that are not specified will be used."

**Review by jean-jimbo-sonarsource — CHANGES_REQUESTED** [review]
Suggestion
